hyp3lib/make_cogs.py

```diff
@@ -1,45 +1,41 @@
-"""Creates a Cloud Optimized Geotiff from the input geotiff(s)"""
+"""Creates a Cloud Optimized GeoTIFF from the input GeoTIFF(s)"""
 
 import argparse
-import glob
 import logging
 import os
 import shutil
+import sys
+from glob import glob
+from tempfile import NamedTemporaryFile
 
 from osgeo import gdal
 
 
-def cogify_dir(dir="PRODUCT",debug=False,res=30):
-    back = os.getcwd()
-    os.chdir(dir)
-    tmpfile = "tmp_cog_{}.tif".format(os.getpid())
-    for myfile in glob.glob("*.tif"):
-        logging.info("Converting file {} into COG".format(myfile))
-        make_cog(myfile,tmpfile,res=res)
-        shutil.move(tmpfile,myfile)
-    os.chdir(back)
+def cogify_dir(directory: str = "PRODUCT", file_pattern: str = '*.tif'):
+    """
+    Convert all found GeoTIFF files to a Cloud Optimized GeoTIFF inplace
+    Args:
+        directory: directory to search through
+        file_pattern: the pattern for finding GeoTIFFs
+    """
+    path_expression = os.path.join(directory, file_pattern)
+    logging.info(f'Converting files to COGs for {path_expression}')
+    for filename in glob(path_expression):
+        make_cog(filename)
 
 
-def make_cog(inFile,outFile,debug=False,res=30):
-    print("Creating COG file {} from input file {}".format(outFile, inFile))
-    tmpFile = 'cog_{}.tif'.format(os.getpid())
-    shutil.copy(inFile,tmpFile)
+def make_cog(filename: str):
```

Comment:
and while we're making breaking changes this might as well be `cogify_file`

Reply:
I agree; done